src/utils/StorageUtils.ts

```diff
@@ -0,0 +1,62 @@
+export type ValidStorageProviders = 'local' | 'session';
+
+export class StorageUtils<T> {
+  private storage: Storage;
+
+  constructor(public id: string, public storageProvider: ValidStorageProviders = 'local') {
+    // This check must be made in a try/catch. If cookies are disabled for a
+    // browser then window.localStorage will throw an undefined exception.
+    try {
+      switch (storageProvider) {
+        case 'local':
+          this.storage = window.localStorage;
+          break;
+        case 'session':
+          this.storage = window.sessionStorage;
+          break;
+        default:
+          this.storage = null;
+      }
+    } catch (error) {
+      this.storage = null;
+    }
+  }
+
+  public save(data: T) {
```

Comment:
Hmm, one think to think about is how we delete the information on the last viewed result stored in session storage in order to allow implementers to be GDPR compliant. We added `clearLocalData` in `Analytics.ts` a couple months ago, and I see that it does not take into account information stored in local/session storage. We could do this in a separate PR though.